### PR TITLE
Fixes the ancient suit from makings stomping noises in peoples hands

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Fixes https://github.com/tgstation/tgstation/issues/28825

### CODE REVIEW

**Comment by Steelpoint** [comment]
Good stuff

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -565,7 +565,10 @@
 	resistance_flags = FIRE_PROOF
 	var/footstep = 1
 
-obj/item/clothing/suit/space/hardsuit/ancient/on_mob_move()
+/obj/item/clothing/suit/space/hardsuit/ancient/on_mob_move()
+	var/mob/living/carbon/human/H = loc
+	if(!ishuman(H) || H.wear_suit != src)
```
Comment (Cyberboss) [inline code/modules/clothing/spacesuits/hardsuit.dm]: Better to use istype than ishuman here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 2]
```diff
@@ -565,7 +565,10 @@
 	resistance_flags = FIRE_PROOF
 	var/footstep = 1
 
-obj/item/clothing/suit/space/hardsuit/ancient/on_mob_move()
+/obj/item/clothing/suit/space/hardsuit/ancient/on_mob_move()
+	var/mob/living/carbon/human/H = loc
```
Comment (Cyberboss) [inline code/modules/clothing/spacesuits/hardsuit.dm]: should call ..() no?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by lordpidey** [comment]
Won't drones still make stompy sounds while holding it?

File: code/modules/clothing/spacesuits/hardsuit.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/modules/clothing/spacesuits/hardsuit.dm]: I have no idea, ..() wasn't there when I started editing this

**Comment by PraiseRatvar** [comment]
does still also fix the sounds when the person wearing the suit has collapsed/stunned/dead?

**Comment by PraiseRatvar** [comment]
does still also fix the sounds when the person wearing the suit has collapsed/stunned/dead while he is getting pulled?

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**